src/Device.vala

```diff
@@ -30,28 +30,46 @@ public class Sound.Device : GLib.Object {
 
     public signal void removed ();
 
+    // info from card and ports
     public bool input { get; set; default=true; }
-    public uint32 index { get; construct; default=0U; }
-    public string name { get; set; }
+    public string id { get; construct; }
+    public string card_name { get; set; }
+    public uint32 card_index { get; construct; }
+    public string port_name { get; construct; }
     public string display_name { get; set; }
     public string form_factor { get; set; }
+    public Gee.ArrayList<string> profiles { get; set; }
```

Comment:
Where is this initialized? I would expect it to be at least initialized to an empty list in the constructor to avoid any null value access.

Reply:
Here: https://github.com/elementary/switchboard-plug-sound/blob/1d3ccbb04aeb499e41acbdb97e019c87129aa5fa/src/PulseAudioManager.vala#L575
But yeah good point, I'll do that.